### PR TITLE
[Inference API] Semantic text delete inference

### PR DESCRIPTION
**Pull request by maxhniebergall**
PR to reintroduce functionality in [this pr](https://github.com/elastic/elasticsearch/pull/110399) after it had to be [reverted](https://github.com/elastic/elasticsearch/pull/110446) due to performing too much work on the transport thread. This commit  follows the pattern from TransportGetTrainedModelsStatsAction for executing on a utility threadpool. Labeling as non-issue to avoid polluting the change log.

### CODE REVIEW

**Comment by maxhniebergall** [comment]
@elasticmachine merge upstream

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/DeleteInferenceEndpointAction.java [hunk 1]
```diff
@@ -126,12 +142,18 @@ public void writeTo(StreamOutput out) throws IOException {
...
+                out.writeOptionalString(dryRunMessage);
+            }
         }
 
         @Override
         protected void addCustomFields(XContentBuilder builder, Params params) throws IOException {
             super.addCustomFields(builder, params);
             builder.field(PIPELINE_IDS, pipelineIds);
+            builder.field(REFERENCED_INDEXES, indexes);
+            builder.field(DRY_RUN_MESSAGE, dryRunMessage);
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/DeleteInferenceEndpointAction.java:156]:
```suggestion
    if (dryRunMessage != null) {
            builder.field(DRY_RUN_MESSAGE, dryRunMessage);
    }        
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/DeleteInferenceEndpointAction.java [hunk 2]
```diff
@@ -142,6 +164,11 @@ public String toString() {
             for (String entry : pipelineIds) {
                 returnable.append(entry).append(", ");
             }
+            returnable.append(", semanticTextFieldsByIndex: ");
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/DeleteInferenceEndpointAction.java:167]: You can use `org.elasticsearch.common.Strings::toString(ToXContent toXContent)` for a simple toString implementation - `return Strings.toString(this);`

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/SemanticTextInfoExtractor.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class SemanticTextInfoExtractor {
+    private static final Logger logger = LogManager.getLogger(SemanticTextInfoExtractor.class);
+
+    public static Set<String> extractIndexesReferencingInferenceEndpoints(Metadata metadata, Set<String> endpointIds) {
+        assert Transports.assertNotTransportThread("non-trivial nested loops over cluster state structures");
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/SemanticTextInfoExtractor.java:27]: 👍

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**